### PR TITLE
feat(capacity): add stake benchmarks 

### LINKED ISSUE #891: [Capacity] benchmark stake extrinsic
Generate weights for stake extrinsic 
@p5150j add meat ...

### PR DESCRIPTION
**Pull request by enddynayn**
# Goal
Add benchmarks for the "stake" extrinsic.
This allows the weights to be used to measure
the cost of calling the "stake" extrinsic.

Closes #891

### CODE REVIEW

File: scripts/run_benchmark.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ $BENCHMARK \
   --pallet pallet_$pallet \
   --extrinsic "*" \
   --chain="frequency-bench" \
-  --execution wasm \
+  --execution native \
```
Comment (enddynayn) [inline scripts/run_benchmark.sh:36]: remove

File: runtime/frequency/src/lib.rs [hunk 2]
```diff
@@ -308,9 +308,11 @@ impl pallet_capacity::Config for Runtime {
 	type RuntimeEvent = RuntimeEvent;
 	type WeightInfo = pallet_capacity::weights::SubstrateWeight<Runtime>;
 	type Currency = Balances;
-	type MinimumStakingAmount = ConstU128<1>;
+	type MinimumStakingAmount = ConstU128<5>;
 	type TargetValidator = Msa;
 	type MaxUnlockingChunks = ConstU32<4>;
+	#[cfg(feature = "runtime-benchmarks")]
```
Comment (enddynayn) [inline runtime/frequency/src/lib.rs:314]: consider using it for test

File: pallets/capacity/src/tests.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@ use super::*;
 use crate as pallet_capacity;
 use crate::mock::*;
 use frame_support::{assert_noop, assert_ok, traits::WithdrawReasons, BoundedVec};
-use testing_utils::{register_provider, run_to_block, staking_events};
+// use testing_utils::{register_provider, run_to_block, staking_events};
```
Comment (enddynayn) [inline pallets/capacity/src/tests.rs:5]: make add back

File: pallets/capacity/src/mock.rs [hunk 4]
```diff
@@ -148,3 +152,32 @@ pub fn new_test_ext() -> sp_io::TestExternalities {
...
+		if System::block_number() > 1 {
+			System::on_finalize(System::block_number());
+		}
+		System::set_block_number(System::block_number() + 1);
+		System::on_initialize(System::block_number());
+		Capacity::on_initialize(System::block_number());
+	}
+}
+
+pub fn register_provider(target_id: MessageSourceId, name: String) {
```
Comment (enddynayn) [inline pallets/capacity/src/mock.rs:180]: might not need this anymore

File: pallets/capacity/src/benchmarking.rs [hunk 5]
```diff
@@ -1,17 +1,58 @@
...
+		let staking_account = Capacity::get_staking_account_for(caller).unwrap();
+		assert_eq!(
+			staking_account,
+			StakingAccountDetails::<T> {
+				active: BalanceOf::<T>::from(10u32),
+				total: BalanceOf::<T>::from(10u32),
+				unlocking: BoundedVec::default(),
+			}
+		)
+		// add target
```
Comment (enddynayn) [inline pallets/capacity/src/benchmarking.rs:54]: do

File: pallets/capacity/src/benchmarking.rs [hunk 6]
```diff
@@ -1,17 +1,42 @@
...
+		let amount: BalanceOf<T> = T::MinimumStakingAmount::get();
 		let target = 1;
 
+		register_provider::<T>(target, "Foo");
+
 	}: _ (RawOrigin::Signed(caller.clone()), target, amount)
 	verify {
+		assert!(StakingAccountLedger::<T>::contains_key(&caller));
+		assert!(StakingTargetLedger::<T>::contains_key(&caller, target));
+		assert!(CapacityLedger::<T>::contains_key(target));
```
Comment (shannonwells) [inline pallets/capacity/src/benchmarking.rs:39]: did you also want to check for any events here?

Comment (enddynayn) [reply pallets/capacity/src/benchmarking.rs:39]: Ahh, good idea.

**Review by shannonwells — APPROVED** [review]
Just the one comment

**Review by johnmela — APPROVED** [review]

**Pull request merged**